src/_pytest/nodes.py

```diff
@@ -462,6 +462,9 @@ def reportinfo(self) -> Tuple[Union[py.path.local, str], Optional[int], str]:
     @cached_property
     def location(self) -> Tuple[str, Optional[int], str]:
         location = self.reportinfo()
-        fspath = self.session._node_location_to_relpath(location[0])
+        fspath = location[0]
+        if not isinstance(fspath, py.path.local):
+            fspath = py.path.local(fspath)
```

Comment:
This might be triggered via https://github.com/pytest-dev/pytest/blob/156a41a855d28ae9f6d83545af69b51fb26f199e/src/_pytest/python.py#L288-L292.

Reply:
Thanks I was wondering this myself because a cursory look at `nose.py` did not bring anything to light.